src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -154,88 +160,88 @@ base::Expression existanceCheck(const std::string& targetField,
     const auto name = formatHelperName(targetField, rawName, parameters);
 
     const auto dbName = parameters[0].m_value;
-
-    // Get DB
-    // TODO: Fix once KVDB is refactored
-    const auto result = kvdbManager->getHandler(dbName);
-    if (std::holds_alternative<base::Error>(result))
-    {
-        throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", std::get<base::Error>(result).message));
-    }
-
     const std::string successTrace {fmt::format("[{}] -> Success", name)};
-
     const std::string failureTrace {
         fmt::format("[{}] -> Failure: Target field '{}' does not exist or it is not a string", name, targetField)};
 
+    auto resultHandler = kvdbManager->getKVDBHandler(dbName, kvdbScopeName);
+
+    if (std::holds_alternative<base::Error>(resultHandler))
+    {
+        throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", std::get<base::Error>(resultHandler).message));
+    }
+
     return base::Term<base::EngineOp>::create(
         name,
-        [=, kvdb = std::get<kvdb_manager::KVDBHandle>(result), targetField = std::move(targetField)](base::Event event)
+        [=,
+         targetField = std::move(targetField),
+         kvdbHandler = std::get<std::shared_ptr<kvdbManager::IKVDBHandler>>(resultHandler)](base::Event event)
         {
             bool found = false;
-            try // TODO We are only using try for JSON::get. Is correct to
-                // wrap everything?
+            std::optional<std::string> value;
+
+            try
             {
-                const auto value = event->getString(targetField);
-                if (value.has_value())
-                {
-                    if (kvdb->hasKey(value.value()))
-                    {
-                        found = true;
-                    }
-                }
+                value = event->getString(targetField);
             }
             catch (std::exception& e)
             {
                 return base::result::makeFailure(event, failureTrace + ": " + e.what());
             }
 
-            // TODO: is this condition right? shouldn't this condition be: "!checkExist ||
-            // (checkExist && found)" as if "checkExist" is "false" and "found" is "true"
-            // then "truth" is false and the result is a
-            bool truth = checkExist ? found : !found;
-            if (truth)
+            if (value.has_value())
             {
-                return base::result::makeSuccess(event, successTrace);
+                auto result = kvdbHandler->contains(value.value());
+                if (std::holds_alternative<base::Error>(result))
+                {
+                    return base::result::makeFailure(event,
+                                                     failureTrace + ": " + std::get<base::Error>(result).message);
+                }
+
+                found = std::get<bool>(result);
             }
-            else
+
+            if ((shouldMatch && found) || (!shouldMatch && !found))
```

Comment:
This method is used for 2 helpers, the one that matches and the one that does not match, the first condition is for case 1, the second for case 2, in case it is not found and the helper that was used is not match, it must return ok.